### PR TITLE
fix: Integration tests on windows

### LINKED ISSUE #1282: Windows | Integration not working
**Describe the bug**
> A clear and concise description of what the bug is.

Integration tests on windows fails with message:

```

  Error: Failed to read service account credential.
    Error reading credential file from environment variable GOOGLE_APPLICATION_CREDENTIALS, value '***
    ': File does not exist.
     expected:<0> but was:<1>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at IntegrationTests.shouldMatchAndroidSuccessExitCodeAndPattern(IntegrationTests.kt:15)
Finished generating test XML results (0.004 secs) into: /home/runner/work/flank/flank/integration_tests/build/test-results/test

```

I tested on my windows vm and it looks ```echo``` on 
```echo certutil -decode gcloud_key.txt %GCLOUD_DIR%application_default_credentials.json ``` 
cause this problem

**To Reproduce**
> Steps to reproduce the behavior:

1. Run integration tests for windows
2. See error

**Expected behavior**
> A clear and concise description of what you expected to happen.

Integration tests should work on windows.

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #1282 

Integration tests on windows should works

### CODE REVIEW

**Review by piotradamczyk5 — APPROVED** [review]

**Review by Sloox — APPROVED** [review]

**Review by pawelpasterz — APPROVED** [review]

File: test_runner/src/main/kotlin/ftl/config/FtlConstants.kt [hunk 1]
```diff
@@ -69,18 +68,20 @@ object FtlConstants {
...
             useMock -> GoogleCredentials.create(AccessToken("mock", Date()))
             UserAuth.exists() -> UserAuth.load()
-            else -> try {
+            else -> runCatching {
                 GoogleApiLogger.silenceComputeEngine()
                 ServiceAccountCredentials.getApplicationDefault()
-            } catch (e: IOException) {
-                throw FlankGeneralError("Error: Failed to read service account credential.\n${e.message}")
+            }.getOrElse {
+                if (isWindows) GoogleCredentials.fromStream(defaultCredentialPath.toFile().inputStream())
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/config/FtlConstants.kt:83]: What will be the error message when flank will fail reading credentials on Windows as well?

File: test_runner/src/main/kotlin/ftl/config/FtlConstants.kt [hunk 2]
```diff
@@ -69,18 +68,20 @@ object FtlConstants {
     }
 
     val defaultCredentialPath: Path by lazy {
-        Paths.get(System.getProperty("user.home"), ".config/gcloud/application_default_credentials.json")
+        if (isWindows) Paths.get(System.getenv("HOMEPATH"), ".config/gcloud/application_default_credentials.json")
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/config/FtlConstants.kt:71]: How about 

```kotlin
       val homePath = if(isWindows) (System.getenv("HOMEPATH") else System.getProperty("user.home")
       Paths.get(homePath, ".config/gcloud/application_default_credentials.json")
```
?

File: test_runner/src/main/kotlin/ftl/config/FtlConstants.kt (hunk 1 above)
Comment (piotradamczyk5) [reply test_runner/src/main/kotlin/ftl/config/FtlConstants.kt:83]: Probably similiar to previous one from `ServiceAccountCredentials.getApplicationDefault()`

Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/config/FtlConstants.kt]: Thanks! I changed it to throws FlankGeneralException like on non-windows.

File: test_runner/src/main/kotlin/ftl/config/FtlConstants.kt (hunk 2 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/config/FtlConstants.kt]: Changed! Thanks!

**Pull request merged**